docs/cas-server-documentation/installation/Cassandra-ticket-registry.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: default
+title: CAS - Cassandra Ticket Registry
+---
+
+# Cassandra Ticket Registry
+Cassandra integration is enabled by including the following dependency in the WAR overlay:
+
+```xml
+<dependency>
+    <groupId>org.apereo.cas</groupId>
+    <artifactId>cas-server-support-cassandra-ticket-registry</artifactId>
+    <version>${cas.version}</version>
+</dependency>
+```
+
+## Configuration Considerations
+
+There are four core configuration concerns with Cassandra:
+
+1. Keyspace
+2. Ticket cleaning
+3. Object serialization mechanism
+4. Multi datacenter replication
+
+### Keyspace
+We suggest the following Keysapace definition:
+
+- String serialized ticket:
+```cql
+CREATE KEYSPACE cas WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': '3'}  AND durable_writes = true;
+
+CREATE TABLE cas.ticket_cleaner (
+    expiry_type text,
+    date_bucket bigint,
+    id text,
+    PRIMARY KEY ((expiry_type, date_bucket), id)
+);
+
+CREATE TABLE cas.ticketgrantingticket (
+    id text PRIMARY KEY,
+    ticket text
+) WITH default_time_to_live = 5184000;
+
+CREATE TABLE cas.serviceticket (
+    id text PRIMARY KEY,
+    ticket text
+) WITH default_time_to_live = 60;
+
+CREATE TABLE cas.ticket_cleaner_lastrun (
+    id text PRIMARY KEY,
+    last_run bigint
+);
+```
+
+- Binary serialized ticket:
+```cql
+CREATE KEYSPACE cas WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': '3'}  AND durable_writes = true;
+
+CREATE TABLE cas.ticket_cleaner (
+    expiry_type text,
+    date_bucket bigint,
+    id text,
+    PRIMARY KEY ((expiry_type, date_bucket), id)
+);
+
+CREATE TABLE cas.ticketgrantingticket (
+    id text PRIMARY KEY,
+    ticket blob
+) WITH default_time_to_live = 5184000;
+
+CREATE TABLE cas.serviceticket (
+    id text PRIMARY KEY,
+    ticket blob
+) WITH default_time_to_live = 60;
+
+CREATE TABLE cas.ticket_cleaner_lastrun (
+    id text PRIMARY KEY,
+    last_run bigint
+);
+```
+
+### Ticket cleaning
+Cassandra supports [TTL](https://en.wikipedia.org/wiki/Time_to_live). We use this feature to clean up Service Tickets as we know the maximum duration.
+This TTL has to be defined in the schema definition:
+```cql
+CREATE TABLE cas.serviceticket (
+    id text PRIMARY KEY,
+    ticket text
+) WITH default_time_to_live = 60;
+```
+
+For TicketGrantingTickets we follow a different approach. When a new TGT is stored, its id is stored in the ticket_cleaner table, within a bucket based on the ticket expiration time.
+This time bucket is 10 seconds.
+The ticket cleaner will check the last bucket period, and retrieve all the ticket's id for the next bucket in the ticket_cleaner table, then it'll run queries agains the TGT table in order to retrieve the whole ticket, and run the cleaning process itself:
+- check if expired, and if so
+- logout user from services
+- remove ticket
```

Comment:
I wrote this for a previous implementation where a specific Cassandra cleaner was in place. Now we are using the default one, so probably this section is not needed